frontend/src/utilities/useWatchComponents.tsx

```diff
@@ -52,7 +52,7 @@ export const useWatchComponents = (
     }
   }, [forceUpdate, installed]);
 
-  const retComponents = useDeepCompareMemoize<OdhApplication[]>(components);
+  const retComponents = useDeepCompareMemoize<OdhApplication[] | undefined>(components);
```

Comment:
Not possible to be `undefined`.
```suggestion
  const retComponents = useDeepCompareMemoize(components);
```